stl/inc/span

```diff
@@ -0,0 +1,723 @@
+        return begin();
+    }
+
+    _NODISCARD constexpr const_iterator cend() const noexcept {
+        return end();
+    }
+
+    _NODISCARD constexpr reverse_iterator rbegin() noexcept {
+        return reverse_iterator{end()};
+    }
+
+    _NODISCARD constexpr reverse_iterator rend() noexcept {
+        return reverse_iterator{begin()};
+    }
+
+    _NODISCARD constexpr const_reverse_iterator rbegin() const noexcept {
+        return const_reverse_iterator{end()};
+    }
+
+    _NODISCARD constexpr const_reverse_iterator rend() const noexcept {
+        return const_reverse_iterator{begin()};
+    }
+
+    _NODISCARD constexpr const_reverse_iterator crbegin() const noexcept {
+        return rbegin();
+    }
+
+    _NODISCARD constexpr const_reverse_iterator crend() const noexcept {
+        return rend();
+    }
+
+    _NODISCARD constexpr pointer _Unchecked_begin() noexcept {
+        return data();
+    }
+
+    _NODISCARD constexpr const_pointer _Unchecked_begin() const noexcept {
+        return data();
+    }
+
+    _NODISCARD constexpr pointer _Unchecked_end() noexcept {
+        return data() + size();
+    }
+
+    _NODISCARD constexpr const_pointer _Unchecked_end() const noexcept {
+        return data() + size();
+    }
+
+    constexpr void swap(span& _Other) noexcept {
+        _Mydata.swap(_Other._Mydata);
+    }
+
+private:
+    _Span_storage<pointer, _Extent> _Mydata;
+};
+
+// [span.tuple] Tuple interface
+template <class _Ty, size_t _Extent>
+struct tuple_size<span<_Ty, _Extent>> : integral_constant<size_t, _Extent> {};
+
+template <class _Ty>
+struct tuple_size<span<_Ty, dynamic_extent>>;
+
+template <size_t _Index, class _Ty, size_t _Extent>
+struct tuple_element<_Index, span<_Ty, _Extent>> {
+    static_assert(dynamic_extent != _Extent, "std::tuple_element<> not supported for std::span<T, dynamic_extent>");
+    static_assert(_Index < _Extent, "Index out of bounds in std::tuple_element<> (std::span)");
+    using type = _Ty;
+};
+
+template <size_t _Index, class _Ty, size_t _Extent>
+_NODISCARD constexpr _Ty& get(span<_Ty, _Extent> _Span) noexcept {
+    static_assert(dynamic_extent != _Extent, "std::get<> not supported for std::span<T, dynamic_extent>");
+    static_assert(_Index < _Extent, "Index out of bounds in std::get<> (std::span)");
+    return _Span[_Index];
+}
+
+// [span.objectrep] Views of object representation
+// FUNCTION TEMPLATE as_bytes
+template <class _Ty, size_t _Extent>
+_NODISCARD auto as_bytes(span<_Ty, _Extent> _Sp) noexcept
+    -> span<byte, _Extent == dynamic_extent ? dynamic_extent : sizeof(_Ty) * _Extent> {
+    return {reinterpret_cast<byte*>(_Sp.data()), _Sp.size_bytes()};
+}
+
+// FUNCTION TEMPLATE as_writable_bytes
+template <class _Ty, size_t _Extent>
+_NODISCARD auto as_bytes(span<_Ty, _Extent> _Sp) noexcept
+    -> enable_if_t<!is_const_v<_Ty>, span<byte, _Extent == dynamic_extent ? dynamic_extent : sizeof(_Ty) * _Extent>> {

```

Comment:
We prefer template-parameter SFINAE to return-type SFINAE since it has less drastic effects on tooltips in the IDE.

Reply:
Done